platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/AddressBookInitializer.java

```
@@ -0,0 +1,274 @@
+     * and the saved state from disk.
+     *
+     * @param currentVersion       The current version of the application. Must not be null.
+     * @param signedState          The signed state loaded from disk.  May be null.
+     * @param genesisSupplier      The swirld application state in genesis start. Must not be null.
+     * @param configAddressBook    The address book derived from config.txt. Must not be null.
+     * @param parentDirectory      The parent directory of the address book directory. Must not be null.
+     * @param useConfigAddressBook Indicates if the unmodified config address book should be used.
+     */
+    public AddressBookInitializer(
+            @NonNull final SoftwareVersion currentVersion,
+            @Nullable final SignedState signedState,
+            @NonNull final Supplier<SwirldState> genesisSupplier,
+            @NonNull final AddressBook configAddressBook,
+            @NonNull final String parentDirectory,
+            @NonNull final boolean useConfigAddressBook) {
+        this.currentVersion = Objects.requireNonNull(currentVersion, "The currentVersion must not be null.");
+        this.genesisSupplier =
+                Objects.requireNonNull(genesisSupplier, "The genesis swirldState supplier must not be null.");
+        this.configAddressBook = Objects.requireNonNull(configAddressBook, "The configAddressBook must not be null.");
+        Objects.requireNonNull(parentDirectory, "The parentDirectory must not be null.");
+        this.useConfigAddressBook = useConfigAddressBook;
+        Path addressBookDirectoryPath = Path.of(parentDirectory, ADDRESS_BOOK_DIRECTORY_NAME);
+        try {
+            Files.createDirectories(addressBookDirectoryPath);
+        } catch (final IOException e) {
+            logger.error(EXCEPTION.getMarker(), "Not able to create directory: {}", addressBookDirectoryPath, e);
+            addressBookDirectoryPath = null;
```

should we proceed with this being null?

---

It is null only when there is an IOException.  Fail hard and throw a runtime exception?  IllegalState?

---

Now throwing an IllegalStateException